server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldValueRetriever.java

```diff
@@ -38,34 +40,40 @@
  * Then given a specific document, it can retrieve the corresponding fields from the document's source.
  */
 public class FieldValueRetriever {
-    private final MappingLookup fieldMappers;
     private final List<FieldContext> fieldContexts;
 
     public static FieldValueRetriever create(MapperService mapperService,
                                              Collection<FieldAndFormat> fieldAndFormats) {
         MappingLookup fieldMappers = mapperService.documentMapper().mappers();
-        List<FieldContext> fields = new ArrayList<>();
+        List<FieldContext> fieldContexts = new ArrayList<>();
 
         for (FieldAndFormat fieldAndFormat : fieldAndFormats) {
             String fieldPattern = fieldAndFormat.field;
             String format = fieldAndFormat.format;
 
             Collection<String> concreteFields = mapperService.simpleMatchToFullName(fieldPattern);
             for (String field : concreteFields) {
-                if (fieldMappers.getMapper(field) != null && mapperService.isMetadataField(field) == false) {
-                    Set<String> sourcePath = mapperService.sourcePath(field);
-                    fields.add(new FieldContext(field, sourcePath, format));
+                Mapper mapper = fieldMappers.getMapper(field);
+                if (mapper == null || mapperService.isMetadataField(field)) {
+                    continue;
+                }
+
+                if (mapper instanceof FieldAliasMapper) {
```

Comment:
This is an unfortunate workaround. I think we should consider moving value fetching support from `FieldMapper` to `MappedFieldType` in a follow-up. That would clean this up, because when looking up a field's `MappedFieldType` we will already resolve aliases to their targets (and can avoid doing it manually here).

Reply:
:+1: